ProteGO/Common/BeaconIdsManager/BeaconIdsManager.swift

```diff
@@ -0,0 +1,70 @@
+import Foundation
+import RealmSwift
+
+final class BeaconIdsManager: BeaconIdsManagerType {
+
+    var allBeaconIds: Results<RealmExpiringBeacon> {
+        return self.realmManager.realm.objects(RealmExpiringBeacon.self).sorted(byKeyPath: "expirationDate", ascending: true)
+    }
+
+    var lastStoredExpiringBeaconDate: Date? {
+        return self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .sorted(byKeyPath: "expirationDate", ascending: false)
+            .first?
+            .expirationDate
+    }
+
+    var currentExpiringBeaconId: ExpiringBeaconId? {
+        let currentDate = currentDateProvider.currentDate
+
+        guard let previousExpiringBeacon = self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .filter("expirationDate <= %@", currentDate).sorted(byKeyPath: "expirationDate", ascending: false).first else {
+                // no valid beacon found
+                return nil
+        }
+
+        guard let previousBeacon = BeaconId(data: previousExpiringBeacon.beaconIdData) else {
+            logger.error("Incorrect beacon found \(previousExpiringBeacon.beaconIdData.toHexString())")
+            return nil
+        }
+
+        guard let nextExpiringBeacon = self.realmManager.realm.objects(RealmExpiringBeacon.self)
+            .filter("expirationDate > %@", currentDate).sorted(byKeyPath: "expirationDate", ascending: true).first else {
+                // last beacon with current date + default lifespan
+                let date = currentDate.addingTimeInterval(Constants.Bluetooth.ExpiringBeaconDefaultLifespan)
+                return ExpiringBeaconId(beaconId: previousBeacon,
+                                        expirationDate: date,
+                                        currentDateProvider: self.currentDateProvider)
+        }
+
+        // beacon with date adjusted to the next available beacon
+        return ExpiringBeaconId(beaconId: previousBeacon,
+                                expirationDate: nextExpiringBeacon.expirationDate,
```

Comment:
Nazwa pola `expirationDate` w `ExpiringBeaconId` jest intuicyjna, ale w `RealmExpiringBeacon` proponuję zmienić na `startDate`, albo coś innego informującego o tym, że to data planowanego początku a nie końca rozgłaszania.

Reply:
👍